confluent_platform/datadog_checks/confluent_platform/data/metrics.yaml

```diff
@@ -0,0 +1,1276 @@
+        # Size of the request queue. A congested request queue will not be able to process incoming or outgoing requests
+        - kafka.network:type=RequestChannel,name=RequestQueueSize
+        # Average fraction of time the network processor threads are idle. Values are between 0 (all resources are used) and 1 (all resources are available)
+        - kafka.network:type=SocketServer,name=NetworkProcessorAvgIdlePercent
+        # Number of requests waiting in the fetch purgatory. This is high if consumers use a large value for fetch.wait.max.ms .
+        - kafka.server:type=DelayedOperationPurgatory,name=PurgatorySize,delayedOperation=Fetch
+        # Number of requests waiting in the producer purgatory. This should be non-zero when acks=all is used on the producer.
+        - kafka.server:type=DelayedOperationPurgatory,name=PurgatorySize,delayedOperation=Produce
+        # Maximum lag in messages between the follower and leader replicas. This is controlled by the `replica.lag.max.messages` config.
+        - kafka.server:type=ReplicaFetcherManager,name=MaxLag,clientId=Replica
+        # Number of leaders on this broker. This should be mostly even across all brokers. If not, set auto.leader.rebalance.enable to true on all brokers in the cluster.
+        - kafka.server:type=ReplicaManager,name=LeaderCount
+        # Number of partitions on this broker. This should be mostly even across all brokers.
+        - kafka.server:type=ReplicaManager,name=PartitionCount
+        # Number of partitions whose in-sync replicas count is less than minIsr.
+        - kafka.server:type=ReplicaManager,name=UnderMinIsrPartitionCount
+        # Number of under-replicated partitions (| ISR | < | all replicas |). Alert if value is greater than 0.
+        - kafka.server:type=ReplicaManager,name=UnderReplicatedPartitions
+      attribute:
+        Value:
+          alias: $domain.$type.$name
+          metric_type: gauge
+
+  # Kafka Broker Controller Metrics (Yammer Metric Gauge)
+  - include:
+      domain: kafka.controller
+      bean:
+        # Number of partitions that don’t have an active leader and are hence not writable or readable. Alert if value is greater than 0.
+        - kafka.controller:type=KafkaController,name=OfflinePartitionsCount
+        # Number of active controllers in the cluster. Alert if the aggregated sum across all brokers in the cluster is anything other than 1 because there should be exactly one controller per cluster.
+        - kafka.controller:type=KafkaController,name=ActiveControllerCount
+      attribute:
+        Value:
+          alias: $domain.$name
+          metric_type: gauge
+
+  # Kafka Broker Server Metrics Regex (Yammer Metric Gauge)
+  - include:
+      domain: kafka.server
+      bean_regex:
+        # Lag in number of messages per follower replica. This is useful to know if the replica is slow or has stopped replicating from the leader.
+        - kafka\.server:type=FetcherLagMetrics,name=ConsumerLag,.*
```

Comment:
The `.*` is for the replica? Can it be added as a tag?

Reply:
> The `.*` is for the replica?

That for matching the rest of the bean name.

```
kafka.server:type=FetcherLagMetrics,name=ConsumerLag,clientId=([-.\w]+),topic=([-.\w]+),partition=([0-9]+)
```
I'm adding ^ as comment for reference.

> Can it be added as a tag?

By default, all properties are used as tags by jmxfetch. Here the metric will be tagged by `type`, `name`, `client_id`, `topic` and `partition`.

Reply:
Oh I missed the `default` behavior 👍